react/src/hooks/useMemoWithPrevious.tsx

```diff
@@ -0,0 +1,36 @@
+import { DependencyList, useEffect, useMemo, useRef, useState } from 'react';
+
+export const useMemoWithPrevious = <T,>(
+  factory: () => T,
+  { deps, initialPrev }: { deps: DependencyList; initialPrev?: T },
+) => {
+  const prevRef = useRef(initialPrev);
+  const [prevResetKey, setPrevResetKey] = useState({});
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const current = useMemo(factory, deps);
+  const memoizedPrev = useMemo(() => {
+    return prevRef.current;
+    // Only update when the reset key changes and deps change
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
```suggestion
```

Reply:
This useMemo does not include `prevRef` as a dependency item because, in this case, only `deps` and `prevResetKey` are intentionally considered as dependency items.
Without this comment, ESLint complains about the missing dependency.